staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -877,8 +896,11 @@ func (r *Request) request(ctx context.Context, fn func(*http.Request, *http.Resp
 		r.retry.After(ctx, r, resp, err)
 
 		done := func() bool {
-			defer readAndCloseResponseBody(resp)
+			// make sure we always drain and close
+			closer := &responseCloser{resp: resp}
+			defer closer.DrainAndClose()
 
+			// TODO: we have to do this *before* IsNextRetry or the response body could be drained/closed when we return ...
```

Comment:
this is the place I don't see a simple way to 1) drain the response body before Seeking to retry, 2) only call `fn` once, and 3) still retry if the seek fails